library/errata_tool_product_version.py

```diff
@@ -293,21 +290,20 @@ def run_module():
         name=dict(required=True),
         description=dict(required=True),
         rhel_release_name=dict(required=True),
-        sig_key_name=dict(default='redhatrelease2'),
+        sig_key_name=dict(default="redhatrelease2"),
         default_brew_tag=dict(required=True),
-        is_server_only=dict(type='bool', required=True),
-        enabled=dict(type='bool', default=True),
-        allow_rhn_debuginfo=dict(type='bool', required=True),
-        allow_buildroot_push=dict(type='bool', required=True),
-        is_oval_product=dict(type='bool', required=True),
-        is_rhel_addon=dict(type='bool', required=True),
-        push_targets=dict(type='list', required=True),
-        brew_tags=dict(type='list', required=True),
-    )
-    module = AnsibleModule(
-        argument_spec=module_args,
-        supports_check_mode=True
+        is_server_only=dict(type="bool", required=True),
+        enabled=dict(type="bool", default=True),
+        allow_rhn_debuginfo=dict(type="bool", required=True),
+        allow_buildroot_push=dict(type="bool", required=True),
+        is_oval_product=dict(type="bool", required=True),
+        is_rhel_addon=dict(type="bool", required=True),
+        push_targets=dict(type="list", required=True),
+        brew_tags=dict(type="list", required=True),
+        use_quay_for_containers=dict(type="bool", default=False),
```

Comment:
Please take out the `default` values (`default=False`). This is a pattern that I wish I had not used early on here. It means that if a playbook author omits the `use_quay_for_containers` parameter on an `errata_tool_product_version` task, Ansible will surprisingly reset the value of `use_quay_for_containers` to `False` on the server, even if it was set to `True` on the server previously.

Instead, you can let it simply default to `None`. Later on, right before you call `ensure_product_version()`, if the value is indeed `None`, then drop it from `params`. Like this:

```python
    # `use_quay_for_containers` is optional - don't modify the server-side value if the user has omitted it.
    if params['use_quay_for_containers'] is None:
        params.pop('use_quay_for_containers')
```